pkgs/development/libraries/agda/generic/default.nix

```diff
@@ -0,0 +1,37 @@
+{ lib, mkDerivation, fetchFromGitHub, standard-library }:
+
+mkDerivation rec {
+  pname = "generic";
+  version = "0.1";
+
+  src = fetchFromGitHub {
+    repo = "Generic";
+    owner = "effectfully";
+    rev = "${version}";
+    sha256 = "04a878g0af0pkaxasz4x6cqqh5y0js5d9pv7vf6dd0dq3w0qn1ks";
+  };
+
+  buildInputs = [
+    standard-library
+  ];
+
+  # # everythingFile = "src/Generic/Main.agda";
+  # # No Everything.agda file
+  # buildPhase = "
+  #   find -name '*.agda' -exec agda {} \\;
+  # ";
+
+  preBuild = ''
+    echo "module Everything where" > Everything.agda
+	  find src -name '*.agda' -and -not -name '*Lookup*' | sed -e 's/src\///;s/\//./g;s/\.agda$//;s/^/import /' >> Everything.agda
```

Comment:
That's amazing. Maybe make this a function/option in the nixpkgs agda lib?

Reply:
It could also be a function that computes a new derivation, given an agda `src`. In that case, it's possible to patch the automatically generated `Everything` file.

Reply:
I shamelessly stole this from https://github.com/UlfNorell/agda-prelude/blob/master/test/Makefile but I agree it could be made to work in general